extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/support/FastCamelContext.java

```diff
@@ -331,26 +331,27 @@ protected ReactiveExecutor createReactiveExecutor() {
         return new ReactiveExecutorResolver().resolve(this);
     }
 
-    public AsyncProcessor createMulticast(Collection<Processor> processors, ExecutorService executor,
-            boolean shutdownExecutorService) {
-        return new MulticastProcessor(this, processors, (AggregationStrategy) null, true, executor, shutdownExecutorService,
-                false, false, 0L, (Processor) null, false, false);
+    @Override
+    public AsyncProcessor createMulticast(Collection<Processor> processors, ExecutorService executor, boolean shutdownExecutorService) {
+        return new MulticastProcessor(this, processors, null, true, executor, shutdownExecutorService,
+                false, false, 0L, null, false, false);
     }
 
+    @SuppressWarnings("unchecked")
     protected <T> T resolve(Class<T> clazz, String type, String name, CamelContext context) {
         T result = context.getRegistry().lookupByNameAndType(name, clazz);
-        if (result instanceof CamelContextAware) {
-            ((CamelContextAware) result).setCamelContext(context);
-        }
-        PropertiesComponent comp = getPropertiesComponent();
-        if (comp instanceof org.apache.camel.component.properties.PropertiesComponent) {
-            Properties props = ((org.apache.camel.component.properties.PropertiesComponent) comp).getInitialProperties();
-            if (props != null) {
-                String pfx = CamelRuntime.PFX_CAMEL + type + "." + name;
-                log.debug("Binding {} {} with prefix {}", type, name, pfx);
-                RuntimeSupport.bindProperties(this, props, result, pfx);
-            }
+
+        CamelContextAware.trySetCamelContext(result, context);
+
+        Properties props = getPropertiesComponent().loadProperties();
+        if (props != null && !props.isEmpty()) {
+            PropertyBindingSupport.bindProperties(
+                this,
+                result,
+                new HashMap<String, Object>((Map) props),
```

Comment:
Is there any good reason to allocate a new `HashMap` ? Simply `(Map) props` would compile too.

Reply:
no, leftover from another solution that I had before

Reply:
fixed

Reply:
@ppalaga 

ah no sorry, now I recall the reason: `PropertyBindingSupport.bindProperties` consumes the map so it removes every element that has been bound to the target leading to unexpected results.

pushed a new version that restores hash map, I need to work on the camel side to have a better solution but in the meantime I would merge the PR as it is as `PropertyBindingSupport` to support OGNL like property setting.